### PR TITLE
GCS_MAVLink: Added sanity check for DO_JUMP command on mission load.

### PR DESCRIPTION
**Pull request by nitirohilla**
This PR resolves the issue #4304

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -817,6 +817,14 @@ bool GCS_MAVLINK::handle_mission_item(mavlink_message_t *msg, AP_Mission &missio
         result = MAV_MISSION_INVALID_SEQUENCE;
         goto mission_ack;
     }
+
+    // sanity check for DO_JUMP command
+    if (cmd.id == MAV_CMD_DO_JUMP) {
+        if (cmd.content.jump.target >= waypoint_request_last || cmd.content.jump.target == 0) {
```
Comment (WickedShell) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: I don't think this is valid as it doesn't take into account partial mission uploads. IE I can have 100 waypoints uploaded, and want to swap waypoint 40-50 out with 41 being a jump to 90. waypoint_request_last will be set to 50 in this case and we would reject a valid command. This check should instead be based around checking if the command is >= waypoint_request_last or if its >= the mission total number of commands.

**Comment by WickedShell** [comment]
Other then the comment about partial missions it looks good to me. Applying this in GCS_MAVLink rather then AP_Mission is correct, because AP_Mission is unaware of what the state of the uploading of a mission is. (Which it really should be made to behave that way in the future).

**Comment by nitirohilla** [comment]
@WickedShell I have update the PR taking into consideration partial mission request as per your suggestion.

**Comment by WickedShell** [comment]
You don't need the extra tracking of if it's a partial upload. Because an upload will shorten the number of mission commands that are coming in you should be able to make it a single check of ((cmd.content.jump.target >= mission.num_commands() && cmd.content.jump.target >= waypoint_request_last) || cmd.content.jump.target == 0))

**Comment by nitirohilla** [comment]
Sorry for misunderstanding... Updated the PR :-)

**Comment by WickedShell** [comment]
@nitirohilla I want to apologize somehow I completely lost track of this PR!

I'm good with the changes now. @tridge @rmackay9 or @magicrub can one of you take some time to look at this and pull it in?

**Comment by magicrub** [comment]
closed via https://github.com/ArduPilot/ardupilot/commit/cadd95fdd859e6dac21eaa37f58f057113bb3344, thanks!

**Pull request closed without merging**